tableschema/__init__.py

```diff
@@ -8,6 +8,7 @@
 # Module API
 
 from .schema import Schema
+from .schema import FailedCast
```

Comment:
I don't fully understand the purpose of `FailedCast`. 

It's not used in the code (only in tests) but added to the public API. My guess that clients are going to use it to embrace this new feature. If yes, please add to the `README.md` (I guess somewhere around `Schema/Field`)

Reply:
The intention of FailedCast is to be able to yield rows containing values that weren't actually castable, instead of suppressing the rows completely. It is meant as a means to recognize such non-successfully-casted values by the consumer of the yielded values. This is why they are only used in the tests.
In a pure-validation scenario I presume you don't actually use the yielded rows, they are basically ignored. 

Sidenote:
Thinking about that a bit more if you validated on-the-fly *and* want to immediately process the data you'd probably want something like an additional "iter_validated"-method that yields both

- the data rows and
- the validation errors

e.g. as a (named)tuple.

I.e. in sync, whereas the exc_handler mechanism I propose is more of an asynchronous beast (you iter and later check the aggregated validation errors).
But I think s.th. like iter_validate could be easily implemented on top of iter with exc_handler...

Reply:
FailedCast is now documented, both in the custom exc. handler intro in the Table class section and as a reference (sub) section.